apps/docs-snippets/src/guide/contracts/deploying-contracts.test.ts

```diff
@@ -1,65 +1,110 @@
+    const contractSizeLimit = consensusParameters.contractParameters.contractMaxSize;
+    // #endregion get-contract-max-size
+    expect(contractSizeLimit).toBeDefined();
+  });
 
-    // #region contract-setup-4
+  it('deploys a contract', async () => {
+    using launched = await launchTestNode();
+
+    const {
+      provider: testProvider,
+      wallets: [testWallet],
+    } = launched;
+    const TESTNET_NETWORK_URL = testProvider.url;
+    const WALLET_PVT_KEY = testWallet.privateKey;
+    const bytecode = TypegenFactory.bytecode;
+
+    // #region setup
+    // #import { Provider, TESTNET_NETWORK_URL, Wallet };
+    // #context import { WALLET_PVT_KEY } from 'path/to/my/env/file';
+    // #context import { TypegenFactory } from 'path/to/typegen/outputs';
+
+    const provider = await Provider.create(TESTNET_NETWORK_URL);
+    const wallet = Wallet.fromPrivateKey(WALLET_PVT_KEY, provider);
+    const factory = new TypegenFactory(wallet);
+    // #endregion setup
+    expect(hexlify(factory.bytecode)).toBe(hexlify(bytecode));
+
+    // #region deploy
+    // Deploy the contract
+    const { waitForResult, contractId, transactionId } = await factory.deploy();
+    // Await it's deployment
     const { contract, transactionResult } = await waitForResult();
-    // #endregion contract-setup-4
-
-    // #region contract-setup-5
-    const call = await contract.functions.echo_u8(15).call();
+    // #endregion deploy
+    expect(contract).toBeDefined();
+    expect(transactionId).toBeDefined();
+    expect(transactionResult.status).toBeTruthy();
+    expect(contractId).toBe(contract.id.toB256());
+
+    // #region call
+    // Call the contract
+    const { waitForResult: waitForCallResult } = await contract.functions.echo_u8(10).call();
+    // Await the result of the call
+    const { value } = await waitForCallResult();
+    // #endregion call
+    expect(value).toBe(10);
+  });
 
-    const { value } = await call.waitForResult();
-    // #endregion contract-setup-5
+  it('deploys a large contract as blobs', async () => {
+    using launched = await launchTestNode();
+
+    const {
+      provider: testProvider,
+      wallets: [testWallet],
+    } = launched;
+    const TESTNET_NETWORK_URL = testProvider.url;
+    const WALLET_PVT_KEY = testWallet.privateKey;
+    const abi = EchoValues.abi;
+    const bytecode = TypegenFactory.bytecode;
+
+    // #region blobs
+    // #import { Provider, TESTNET_NETWORK_URL, Wallet, ContractFactory };
+    // #context import { WALLET_PVT_KEY } from 'path/to/my/env/file';
+    // #context import { bytecode, abi } from 'path/to/typegen/outputs';
+
+    const provider = await Provider.create(TESTNET_NETWORK_URL);
+    const wallet = Wallet.fromPrivateKey(WALLET_PVT_KEY, provider);
+    const factory = new ContractFactory(bytecode, abi, wallet);
+
+    // Deploy the contract as blobs
+    const { waitForResult, contractId, transactionId } = await factory.deployContractAsBlobs({
```

Comment:
Have we added a test that covers the functionality of the `deploy` function that automatically decides between a 'normal' deployment and a blob tx depending on the contract's size?

Reply:
Covered in the `fuel-gauge` contract factory tests but I've also added [this](https://github.com/FuelLabs/fuels-ts/pull/2827/commits/4cb5ff6ee44c396dcff9d638c8d58fbd432c5ff3) to check bytecode len.